File: Onchain-data/odsbaron/liqudity_monitor/internal/morpho/abi.go
package morpho

import (
	"strings"

	"github.com/ethereum/go-ethereum/accounts/abi"
	"github.com/ethereum/go-ethereum/crypto"
)

const morphoABIJSON = `[
  {
    "inputs": [{"internalType": "bytes32", "name": "id", "type": "bytes32"}],
    "name": "idToMarketParams",
    "outputs": [
      {"internalType": "address", "name": "loanToken", "type": "address"},
      {"internalType": "address", "name": "collateralToken", "type": "address"},
      {"internalType": "address", "name": "oracle", "type": "address"},
      {"internalType": "address", "name": "irm", "type": "address"},
      {"internalType": "uint256", "name": "lltv", "type": "uint256"}
    ],
    "stateMutability": "view",
    "type": "function"
  },
  {
    "anonymous": false,
    "inputs": [
      {"indexed": true, "internalType": "bytes32", "name": "id", "type": "bytes32"},
      {"indexed": true, "internalType": "address", "name": "caller", "type": "address"},
      {"indexed": true, "internalType": "address", "name": "borrower", "type": "address"},
      {"indexed": false, "internalType": "uint256", "name": "repaidAssets", "type": "uint256"},
      {"indexed": false, "internalType": "uint256", "name": "repaidShares", "type": "uint256"},
      {"indexed": false, "internalType": "uint256", "name": "seizedAssets", "type": "uint256"},
      {"indexed": false, "internalType": "uint256", "name": "badDebtAssets", "type": "uint256"},
      {"indexed": false, "internalType": "uint256", "name": "badDebtShares", "type": "uint256"}
    ],
    "name": "Liquidate",
    "type": "event"
  }
]`

const oracleABIJSON = `[
  {
    "inputs": [],
    "name": "price",
    "outputs": [{"internalType": "uint256", "name": "", "type": "uint256"}],
    "stateMutability": "view",
    "type": "function"
  }
]`

const erc20ABIJSON = `[
  {
    "inputs": [],
    "name": "decimals",
    "outputs": [{"internalType": "uint8", "name": "", "type": "uint8"}],
    "stateMutability": "view",
    "type": "function"
  }
]`

var (
	MorphoABI      = mustParseABI(morphoABIJSON)
	OracleABI      = mustParseABI(oracleABIJSON)
	ERC20ABI       = mustParseABI(erc20ABIJSON)
	LiquidateTopic = crypto.Keccak256Hash([]byte("Liquidate(bytes32,address,address,uint256,uint256,uint256,uint256,uint256)"))
)

func mustParseABI(raw string) abi.ABI {
	parsed, err := abi.JSON(strings.NewReader(raw))
	if err != nil {
		panic(err)
	}
	return parsed
}

File: Onchain-data/RenJW418/Liquidation_Monitor/src/liquidation_sentinel/abi.py
MORPHO_ABI = [
    {
        "anonymous": False,
        "inputs": [
            {"indexed": True, "name": "id", "type": "bytes32"},
            {"indexed": True, "name": "caller", "type": "address"},
            {"indexed": True, "name": "borrower", "type": "address"},
            {"indexed": False, "name": "repaidAssets", "type": "uint256"},
            {"indexed": False, "name": "repaidShares", "type": "uint256"},
            {"indexed": False, "name": "seizedAssets", "type": "uint256"},
            {"indexed": False, "name": "badDebtAssets", "type": "uint256"},
            {"indexed": False, "name": "badDebtShares", "type": "uint256"},
        ],
        "name": "Liquidate",
        "type": "event",
    },
    {
        "inputs": [{"name": "id", "type": "bytes32"}],
        "name": "idToMarketParams",
        "outputs": [
            {"name": "loanToken", "type": "address"},
            {"name": "collateralToken", "type": "address"},
            {"name": "oracle", "type": "address"},
            {"name": "irm", "type": "address"},
            {"name": "lltv", "type": "uint256"},
        ],
        "stateMutability": "view",
        "type": "function",
    },
]

ORACLE_ABI = [
    {
        "inputs": [],
        "name": "price",
        "outputs": [{"name": "", "type": "uint256"}],
        "stateMutability": "view",
        "type": "function",
    }
]

ERC20_ABI = [
    {
        "inputs": [],
        "name": "decimals",
        "outputs": [{"name": "", "type": "uint8"}],
        "stateMutability": "view",
        "type": "function",
    },
    {
        "inputs": [],
        "name": "symbol",
        "outputs": [{"name": "", "type": "string"}],
        "stateMutability": "view",
        "type": "function",
    },
]
